client/packages/common/src/plugins/hooks/usePluginColumns.ts

```diff
@@ -3,21 +3,22 @@ import { ColumnPluginType } from '../types';
 import { RecordWithId } from '../../types/utility';
 import { ColumnDefinition } from '../../ui';
 import { usePluginProvider } from '../components';
+import { ArrayUtils } from '../../utils/arrays';
 
 export function usePluginColumns<T extends RecordWithId>({
   type,
 }: {
   type: ColumnPluginType;
 }) {
-  const { columnPlugins } = usePluginProvider();
+  const columnPlugins = usePluginProvider(state => state.columnPlugins);
```

Comment:
using `usePluginProvider()` causes the provider and all children to re-render, so a specific fetch is required.